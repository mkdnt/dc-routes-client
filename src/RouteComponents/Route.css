.route-card {
  -webkit-box-shadow: 0px 0px 15px 5px #b40421, 0px 0px 15px 5px #b40421;
  box-shadow: 0px 0px 15px 5px #b40421, 0px 0px 15px 5px #b40421;
  -webkit-border-radius: 16px;
  -moz-border-radius: 16px;
  border-radius: 16px;
  padding: 1px;
  margin: 30px;
}

textarea {
  width: 90%;
  height: 150px;
}

input {
  margin: 5px;
}
