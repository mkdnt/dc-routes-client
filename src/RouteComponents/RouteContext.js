import React from 'react';

export default React.createContext({
    routes: [],
    addRoute: () => {},
    editRoute: () => {},
    deleteRoute: () => {},
    editing: null
})