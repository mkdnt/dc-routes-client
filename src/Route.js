import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import RouteContext from './RouteContext'

export class Route extends Component {

static contextType = RouteContext
    render() {
        const { id, name, area, distance, type } = this.props
        return (
            <div>
                <Link 
                to={`/route/${id}`}>
                <h4>{name}</h4>
                </Link>
                <h5>{area}</h5>
                <h5>{distance} miles</h5>
                <h5>{type}</h5>
                <br />
            </div>
        )
    }
}

export default Route