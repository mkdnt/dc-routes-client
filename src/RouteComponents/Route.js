import React, { Component } from 'react'
import RouteContext from './RouteContext'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import './Route.css'


export class Route extends Component {
    
    static propTypes = {
        route: PropTypes.shape({
            id: PropTypes.number.isRequired,
            route_name: PropTypes.string.isRequired,
            dc_area: PropTypes.string.isRequired,
            distance: PropTypes.number.isRequired,
            difficulty: PropTypes.string.isRequired,
            route_type: PropTypes.string.isRequired,
            route_description: PropTypes.string
        })
    }

    static contextType = RouteContext

    render() {
        const { id, name, area, distance, type, difficulty, description } = this.props
        return (
            <div className='route-card'>
                <Link 
                to={`/route/byid/${id}`}
                style={{textDecoration: 'none', color: 'inherit'}}>
                <h3>{name}</h3>
                </Link>
                <h5>{area} | {type}</h5>
                <h5>{distance} miles | Difficulty: {difficulty}</h5>
                <p>{description}</p>
            </div>
        )
    }
}

export default Route