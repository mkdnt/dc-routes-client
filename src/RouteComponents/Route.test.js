import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom';
import RouteContext from './RouteContext';
import Route from './Route';

describe('Route Component', () => {
  describe('Snapshot', () => {
    it('renders the UI as expected', () => {
      const update = renderer.create(<BrowserRouter><Route /></BrowserRouter>)
        expect(update).toMatchSnapshot()
    })
  })
  
  describe('Smoke test', () => {
    it('Renders without crashing', () => {
      const div = document.createElement('div');
      
      const value = {
        routes: [],
        addRoute: () => {},
        editRoute: () => {},
        deleteRoute: () => {}
        }

      ReactDOM.render(
        <BrowserRouter>
          <RouteContext.Provider value={value}>
            <Route />
        </RouteContext.Provider>
        </BrowserRouter>,
      div);
      ReactDOM.unmountComponentAtNode(div);
    })
  })

});