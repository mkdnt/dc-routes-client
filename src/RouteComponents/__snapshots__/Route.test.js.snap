// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Component Snapshot renders the UI as expected 1`] = `
<div
  className="route-card"
>
  <a
    href="/route/byid/undefined"
    onClick={[Function]}
    style={
      Object {
        "color": "inherit",
        "textDecoration": "none",
      }
    }
  >
    <h3 />
  </a>
  <h5>
     | 
  </h5>
  <h5>
     miles | Difficulty: 
  </h5>
  <p />
</div>
`;
